scripts/buildwarns.py

```diff
@@ -29,32 +29,43 @@
         code=line.split()[1].replace(':','')
         percode[code] += 1
         percode['total'] += 1
-        
-        myfile = lastline.split('"')[1]
+        try:
+            myfile = lastline.split('"')[1]
+        except:
+            # Cannot parse line
+            continue
         perfile[myfile] += 1
         if myfile not in fulldata:  # init
             fulldata[myfile] = collections.Counter()
-        fulldata[myfile][code] += 1 
+        fulldata[myfile][code] += 1
         fulldata[myfile]['total'] += 1
     lastline=line
 
 print '=' * 80
 print 'Warnings by type/code'
 for key, value in sorted(percode.iteritems(), reverse=True, key=lambda (k,v): (v,k)):
-    print percode[key] , errorcodes[key]
-
+    try:
+        print percode[key] , errorcodes[key]
+    except KeyError:
+        continue
 print '=' * 80
 print 'Warnings by file'
 for key, value in sorted(perfile.iteritems(), reverse=True, key=lambda (k,v): (v,k)):
-    print value, key
-    
+    try:
+        print value, key
+    except KeyError:
```

Comment:
this and the next try/except don't do anything. print won't fail with a KeyError :)

Reply:
ack, rm'd